src/controllers/UserController.ts

```
@@ -0,0 +1,158 @@
+    const { fullname, email, login, password, isActive } = req.body;
+
+    if (!fullname || !email || !login || !password || isActive === undefined) {
+      return res.status(400).json({ message: 'Fullname, email, login, password, and isActive are required' });
+    }
+
+    try {
+      
+      const hashedPassword = await bcrypt.hash(password, 10);
+      
+      const user = new User();
+      user.fullname = fullname; 
+      user.email = email;
+      user.login = login;
+      user.password = hashedPassword;
+      user.isActive = isActive;
+      
+      const userRepository = dataSource.getRepository(User);
+      const savedUser = await userRepository.save(user);
+      return res.status(201).json(savedUser);
+    } catch (error) {
+      return res.status(500).json({ message: 'Internal Server Error', error });
+    }
+  }
+
+  static async getUsers(req: Request, res: Response) {
+    const { fullname, email, page = 1, limit = 10 } = req.query;
+
+    const userRepository = dataSource.getRepository(User);
+
+    try {
+      const queryBuilder = userRepository.createQueryBuilder('user');
+
+      queryBuilder.where('user.isActive = :isActive', { isActive: true });
+
+      if (fullname) {
+        queryBuilder.andWhere('user.fullname LIKE :fullname', { fullname: `%${fullname}%` });
+      }
+
+      if (email) {
+        queryBuilder.andWhere('user.email LIKE :email', { email: `%${email}%` });
+      }
+
+      queryBuilder.skip((Number(page) - 1) * Number(limit));
+      queryBuilder.take(Number(limit));
+
+      const [users, total] = await queryBuilder.getManyAndCount();
+
+      if (users.length === 0) {
+        return res.status(404).json({ message: 'No users found' });
+      }
+
+      return res.status(200).json({
+        total,
+        page: Number(page),
+        limit: Number(limit),
+        users,
+      });
+    } catch (error) {
+      return res.status(500).json({ message: 'Internal Server Error' });
```

If you want to suppress error messages to the client, you should at least log them, otherwise it would be hard to maintain